app/controllers/concerns/accounts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class AccountsController < ApplicationController
+  def authenticate
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:user_id] = @user.id
+      redirect_to :topics
+    else
+      @errors = ["Hey stupid, enter your information correctly!"]
```

Comment:
This is how we processed errors in phase 2. While you can still do this, you can now make use of Rails' built in "Flash" hash. Everything in the hash is available on the view for the next redirect only. [Docs on Flash.](http://guides.rubyonrails.org/action_controller_overview.html#the-flash)

Note that because you are redirecting here, your `@errors` instance variable will disappear anyway.